frame/support/src/traits/schedule.rs

```diff
@@ -49,86 +49,324 @@ pub const HARD_DEADLINE: Priority = 63;
 /// The lowest priority. Most stuff should be around here.
 pub const LOWEST_PRIORITY: Priority = 255;
 
-/// A type that can be used as a scheduler.
-pub trait Anon<BlockNumber, Call, Origin> {
-	/// An address which can be used for removing a scheduled task.
-	type Address: Codec + Clone + Eq + EncodeLike + Debug;
-
-	/// Schedule a dispatch to happen at the beginning of some block in the future.
-	///
-	/// This is not named.
-	fn schedule(
-		when: DispatchTime<BlockNumber>,
-		maybe_periodic: Option<Period<BlockNumber>>,
-		priority: Priority,
-		origin: Origin,
-		call: Call,
-	) -> Result<Self::Address, DispatchError>;
-
-	/// Cancel a scheduled task. If periodic, then it will cancel all further instances of that,
-	/// also.
-	///
-	/// Will return an error if the `address` is invalid.
-	///
-	/// NOTE: This guaranteed to work only *before* the point that it is due to be executed.
-	/// If it ends up being delayed beyond the point of execution, then it cannot be cancelled.
-	///
-	/// NOTE2: This will not work to cancel periodic tasks after their initial execution. For
-	/// that, you must name the task explicitly using the `Named` trait.
-	fn cancel(address: Self::Address) -> Result<(), ()>;
-
-	/// Reschedule a task. For one-off tasks, this dispatch is guaranteed to succeed
-	/// only if it is executed *before* the currently scheduled block. For periodic tasks,
-	/// this dispatch is guaranteed to succeed only before the *initial* execution; for
-	/// others, use `reschedule_named`.
-	///
-	/// Will return an error if the `address` is invalid.
-	fn reschedule(
-		address: Self::Address,
-		when: DispatchTime<BlockNumber>,
-	) -> Result<Self::Address, DispatchError>;
-
-	/// Return the next dispatch time for a given task.
-	///
-	/// Will return an error if the `address` is invalid.
-	fn next_dispatch_time(address: Self::Address) -> Result<BlockNumber, ()>;
+/// Type representing a call. Can be either the `Call` value itself or the hash of the encoded
+/// `Call`.
```

Comment:
considering the implementation is generic, we can also document it as such: instead of a call it is just a type.